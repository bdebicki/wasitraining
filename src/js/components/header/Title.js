import DATA_URL from '../../enums/data';
import { TITLE } from '../../enums/content';
import { HEADER } from '../../enums/elementHandlers';
import VIEW_TYPES from '../../enums/viewTypes';
import IntroView from '../../views/IntroView';
import { updateComponentByViewType } from '../../utils/updateView';

export default class Title {
	static updateTitleLocation(el) {
		updateComponentByViewType(el, HEADER.TITLE_HEADER_CLASS);
	}

	static backToHome(e) {
		e.preventDefault();

		if (document.querySelector('html').dataset.view === VIEW_TYPES.YEAR) {
			fetch(DATA_URL)
				.then((response) => response.json())
				.then((data) => {
					const intro = new IntroView(data);
					intro.switchToIntoView();
				})
				.catch((error) => {
					console.log(error);
				});
		}
	}

	static renderTitleLink() {
		const a = document.createElement('a');

		a.href = '#home';
		a.classList.add(HEADER.TITLE_LINK_CLASS);
		a.addEventListener('click', Title.backToHome, null);

		return a;
	}

	static render() {
		const title = document.createElement('h1');
		const link = Title.renderTitleLink();

		link.textContent = TITLE;
		title.classList.add(HEADER.TITLE_CLASS);
		title.appendChild(link);
		Title.updateTitleLocation(title);

		return title;
	}
}
