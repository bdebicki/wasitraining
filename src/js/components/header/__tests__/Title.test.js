import cleanDOM from '../../../../../tests/utils/cleanDOM';
import mockView from '../../../../../tests/utils/mockView';
import VIEW_TYPES from '../../../enums/viewTypes';
import Title from '../Title';
import HEADER from '../elementHandlers/header';

afterAll(() => cleanDOM());

describe('test title component', () => {
	it('check dose title component renders correct', () => {
		// when
		const title = Title.render();

		// then
		expect(title).toMatchSnapshot();
	});

	it('check does title on home view has modifier class name', () => {
		// having
		mockView(VIEW_TYPES.INTRO);

		// when
		const title = Title.render();


		// then
		expect(title.classList.contains(HEADER.TITLE_HEADER_CLASS)).toBeTruthy();
	});
});
