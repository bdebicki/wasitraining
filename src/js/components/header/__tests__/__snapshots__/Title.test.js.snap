// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test title component check dose title component renders correct 1`] = `
<h1
  class="pageName"
>
  <a
    class="pageName__link"
    href="#home"
  >
    was it raining at the open'er festival?
  </a>
</h1>
`;
